src/test/memory.rs

```diff
@@ -0,0 +1,172 @@
+// ! This module defines tests any implementation of the MemoryADT interface
+// must pass.
```

Comment:
then why isn't it in `adt.rs`? Also, these tests need to be exported to allow other libraries implementing a memory to use them.